server/applicationset/applicationset.go

```diff
@@ -366,6 +365,40 @@ func (s *Server) ResourceTree(ctx context.Context, q *applicationset.Application
 	return s.buildApplicationSetTree(a)
 }
 
+func (s *Server) Generate(ctx context.Context, q *applicationset.ApplicationSetGenerateRequest) (*applicationset.ApplicationSetGenerateResponse, error) {
```

Comment:
What about RBAC for this call? It may allow to discover cluster secrets, provate repos, etc. Should we restrict this to people with the "create" rbac?

Reply:
Agree - user must have "create" RBAC to see generated apps. I've added a check here: https://github.com/argoproj/argo-cd/blob/4cbb1a9cafecb0efdcfa604e131bd797a16294da/server/applicationset/applicationset.go#L383